source/gui/addonStoreGui/viewModels/store.py

```diff
@@ -632,28 +632,32 @@ def cancelDownloadAddon(self, listItemVM: AddonListItemVM[_AddonStoreModel]):
 		finally:
 			listItemVM.status = getStatus(listItemVM.model, self._filteredStatusKey)
 
-	def cancelInstallAddon(self, listItemVM: AddonListItemVM[_AddonStoreModel]):
+	def _cancelPendingInstallForAddon(self, listItemVM: AddonListItemVM[_AddonStoreModel]):
+		pendingInstallCopy = addonDataManager._downloadsPendingInstall.copy()
+		for addonData, fileDownloaded in pendingInstallCopy:
+			if addonData == listItemVM:
+				addonDataManager._downloadsPendingInstall.remove((addonData, fileDownloaded))
+				listItemVM.status = getStatus(listItemVM.model, self._filteredStatusKey)
+				# Clean up download file
+				try:
+					os.remove(fileDownloaded)
+				except FileNotFoundError:
+					pass  # File not found is benign and expected in some cases

```

Comment:
what cases is this expected?

```suggestion
					log.debugWarning(f"File already removed {fileDownloaded}")
```